### PR TITLE
Use DRA artifacts for beats and ML dependencies

### PR DESCRIPTION
**Pull request by mark-vieira**
Update our build such that we source our Beats and ML artifact dependencies from the DRA artifact repositories.

### CODE REVIEW

File: x-pack/plugin/ml/build.gradle
Comment (mark-vieira) [inline x-pack/plugin/ml/build.gradle:45]: @droberts195 Can you confirm this is _effectively_ what we were doing before? It seems the ML build was also publishing a non-snapshot version or were those artifacts actual build candidates?

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/release-tests

File: x-pack/plugin/ml/build.gradle
Comment (droberts195) [reply x-pack/plugin/ml/build.gradle:45]: The main side effect of this change will be that the "release tests" builds in elasticsearch-ci will now run with snapshot builds of ML.  That is unlikely to make any difference in reality as I cannot think of any time we've made release builds of the ML C++ different to snapshot builds (apart from the version number).  We've never had C++ functionality behind feature flags that's only accessible in snapshot builds for example.

> It seems the ML build was also publishing a non-snapshot version or were those artifacts actual build candidates?

A single ML build was either snapshot or non-snapshot.  But we make sure both are available for any particular version.  The non-snapshot builds are effectively build candidates.  If an emergency release were instigated then the most recent one of those would get incorporated into a unified build candidate.

So I don't think it really matters that much.  What you have at the moment should be fine or changing it to use non-snapshot ML builds in elasticsearch-ci "release tests" would also be fine.  I think the same is probably true for Beats - they appear to regularly publish non-snapshot builds for each branch as part of the DRA process.

File: x-pack/plugin/ml/build.gradle
Comment (droberts195) [inline x-pack/plugin/ml/build.gradle:16]: This was added in #70105 to avoid the risk of somebody uploading a malicious ml-cpp bundle to Maven Central or some other 3rd party repo we search, and then that getting downloaded in preference to what we've put in our own dedicated repo.  Is it possible to keep this functionality?

**Review by droberts195 — COMMENTED** [review]
One thing that's worth being aware of with this change is that it will increase the gap between when ML branches are created on the day after feature freeze and when Elasticsearch can branch.

At the moment this is the sequence:

1. ML branches 8.5 and bumps main to 8.6.0
2. An ML build kicks off on main to build the first 8.6.0 snapshot
3. Elasticsearch can then immediately branch 8.5 and build it by picking up the most recent 8.5.0 ML build that was built on main before the version bump
4. Elasticsearch can then bump the version to 8.6.0 on main and build it using the newly built ML 8.6.0 snapshot
5. At some point later in the day the release team creates the new CI jobs for the 8.5 branch

What will happen in the future is this:

1. ML branches 8.5 and bumps main to 8.6.0
2. An ML build kicks off on main to build the first 8.6.0 snapshot
3. The release team creates new CI infrastructure to build the 8.5 branch
4. An ML build kicks off on the 8.5 branch and builds an ML 8.5.0 snapshot identical to the last one built off main before the version bump _but uploaded to GCS in the 8.5 directory instead of the master directory_
6. Elasticsearch can then branch 8.5 and build it by picking up the first 8.5.0 ML build built on the 8.5 branch after the required CI jobs were created
7. Elasticsearch can also bump the version to 8.6.0 on main and build it using the newly built ML 8.6.0 snapshot

The extra time lag is the time taken to wait for the new CI infrastructure to be created and the reason for it is the distinction the new system makes between 8.5.0 built off the main branch and 8.5.0 built off the 8.5 branch even though they're built from the same commit.

It's not necessarily a problem, as other aspects of the company-wide branching process take a long time now, but something to be aware of for ES point people who might be eager to get started early in the day after a feature freeze.

**Review by breskeby — APPROVED** [review]

File: x-pack/plugin/ml/build.gradle
Comment (mark-vieira) [reply x-pack/plugin/ml/build.gradle:45]: Thanks David. I'll push forward with this then. I think everyone's cadence for releasing staging builds is potentially different with DRA. I wasn't sure folks produced release builds for any given version prior to FF so I wanted to err on the side of caution and stick with snapshots, since they are most likely to be available immediately after a version bump.

File: x-pack/plugin/ml/build.gradle
Comment (mark-vieira) [reply x-pack/plugin/ml/build.gradle]: Yes, I've added this back. It wasn't clear to me how this differed from `content.includeGroup` initially but reading the docs on this DSL cleared that up.

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**
